pennylane/qnn/keras.py

```diff
@@ -54,10 +51,9 @@ class KerasLayer(Layer):
             arguments of the `add_weight()
             <https://www.tensorflow.org/api_docs/python/tf/keras/layers/Layer#add_weight>`__
             method and values being the corresponding specification.
-        batch_idx (Union[Sequence[int], int]): Argument location of the non-trainable inputs for
-            the circuit. This allows batch execution by creating executable circuits for each
-            input example with the same trainable weights. Default ``None``.
-            See :func:`~.pennylane.transforms.batch_input` for more details.
+        split_batches (bool): If True, any batch dimension in the input tensor will be unstacked
+            and split into multiple QNode executions. Otherwise (default), internal PennyLane
+            broadcasting will be used.
```

Comment:
What's the use case for `split_batches=True`?

Reply:
Two main reasons to allow the old behaviour:
1. There are some components of PennyLane (e.g. `qml.math.density_matrix`) that don't support parameter broadcasting yet. Users may still want these components in their ML models.
2. Parameter broadcasting doesn't support more than one batch dimension. The old behaviour supports more than one batch dimension.

Reply:
Once the density_matrix is updated, I would suggest to remove this.

Reply:
I'm not sure whether there are concrete plans to do this currently :thinking:

Reply:
In that case, alternatively we can check if the QNode has a `density_matrix` measurement, and use the old behaviour if it does. The implementation will be a bit messier but at least it'll be less confusing for the user.

Reply:
It should definitely be the responsibility of the device to choose whether or not it has to split a tape with broadcasted parameters up into multiple tapes. 

I wasn't actually aware of this, and I would call it a bug that needs to be fixed. We either need to make density matrix measurements support parameter broadcasting or split up the tape in `Device.batch_transform` if a measurement is a `density_matrix`.

Reply:
Once the new transform infrastructure comes in, users can just apply the relevant transform directly to the QNode, instead of needing a keyword argument here.

Reply:
Removed this keyword argument and all broadcasting logic out of these classes.